core/domain/question_domain.py

```diff
@@ -147,8 +147,41 @@ def validate(self):
             raise utils.ValidationError(
                 'Expected question_data to be a dict, received %s' %
                 self.question_data)
+
```

Comment:
I can't comment higher up, but maybe we should drop self.title, per the questions design proposal?

Also we might need a new QuestionData object that subclasses State (see other comment). Need to implement some mechanism for migration as well.

Reply:
Actually, scrap that -- @BenHenning's comment got me thinking about whether we actually need a separate object here in the first place. Perhaps we can pull the question/exploration-specific validation checks  higher up, so that they're done in exploration.validate() or question.validate() instead? Then outcome.validate() or state.validate() would only do the checks that are common to both cases.

Reply:
That sounds good to me. I was only thinking of this from the questions & shared perspective before, but pulling up exploration-specific checks above the state level also seems reasonable and provides parity with questions.

Reply:
With question title dropped, QuestionSummary has only question_id. So, should that also be removed, or should some values of question_data be added to summary?

Ok, I will do the changes to the validations.

Reply:
Then, for creating a default question_data state, shall I create another function in class State (create_default_question_data()), or for this, should I create the new QuestionData object, like the initial comment, and have this function inside that (in this case, should the interaction and outcome, of question_data, still be objects of their respective classes in exp_domain, or just individual dicts?).

Reply:
Re QuestionSummary: I'm not sure, actually. I haven't fully thought about what QuestionSummary should look like. It would probably contain a summary of the question text but we'll need to figure out how to generate that. For now, I suggest storing the content of the question's state; what do you think?

Re creating a default state, I think a new function in class State is fine. You might even go so far as to rename question_data to 'state' altogether, since it is a state.

Reply:
Ok, I'll create the function in class State.

If the entire question_data is stored, then there is not much difference between class Question and QuestionSummary (only question_id, schema_version and language_code). I thought of just storing interaction name or some field of question_data only, or is a QuestionSummary required, as question object itself doesn't have a lot of information for itself?

Reply:
Done.
Also, I changed the QuestionSummary to just hold question id and content, but when a question is created, there is no function to create its summary as of now (even in develop). Should I add that now or later?